### PR TITLE
Improve n_jobs docs

### LINKED ISSUE #2630: Improve settings documentation
### What kind of feature would you like to request?

Other?

### Please describe your wishes

Hello Scanpy team!

In scanpy api documentation I see [some settings](https://scanpy.readthedocs.io/en/stable/generated/scanpy._settings.ScanpyConfig.html#scanpy._settings.ScanpyConfig), however I don't understand how we are supposed to use n_job and max_memory settings. I would like scanpy to use whatever is available. How can I do that?

### PR DESCRIPTION
**Pull request by flying-sheep**
Fixes #2630

### CODE REVIEW

File: scanpy/_settings.py [hunk 1]
```diff
@@ -303,9 +303,9 @@ def cache_compression(self, cache_compression: Optional[str]):
     @property
     def max_memory(self) -> Union[int, float]:
         """\
-        Maximal memory usage in Gigabyte.
+        Maximum memory usage in Gigabyte.
 
-        Is currently not well respected....
+        Is currently not well respected…
```
Comment (Zethson) [inline scanpy/_settings.py:308]: Should we just remove it all together? It seems to be more confusing than helpful.

File: scanpy/_settings.py [hunk 2]
```diff
@@ -318,6 +318,10 @@ def max_memory(self, max_memory: Union[int, float]):
     def n_jobs(self) -> int:
         """\
         Default number of jobs/ CPUs to use for parallel computing.
+
+        Set to `-1` in order to use all available cores.
+        Not all algorithms support special behavior for numbers < `-1`,
```
Comment (Zethson) [inline scanpy/_settings.py:323]: How about we validate this with code?

**Review by Zethson — APPROVED** [review]
Two optional comments. Feel free to merge

**Comment by flying-sheep** [comment]
I already had automerge active! I’d have added those 😢

**Comment by Zethson** [comment]
Oh sorry, forgot about that. Happy to review a followup :)

**Pull request merged**